### DISCUSSION TITLE
Can DuckDB query Parquet files stored on S3?

### DISCUSSION THREAD

**Discussion by collimarco** [duckdb/duckdb#7841] [General]
In case the answer is yes, does DuckDB need to first download the files **entirely**?

Or it can use predicate pushdown to **exclude the row groups** that don't match the query and thus have only a **partial download**?

**Comment by avyfain** [comment]
AFAICT, in 0.9.0 a basic query does predicate pushdown correctly
```sql
select *
from read_parquet('s3://some/path/*/*/*.parquet', hive_partitioning=True)
where a = 'x' and b='y'
```

a model with a simple CTE does as well
```sql
with my_table as (
    select *
    from read_parquet('s3://some/path/*/*/*.parquet', hive_partitioning=True)
)
   
select *
from my_table
where a = 'x' and b='y'
``` 

but apparently not on a more complex query joining CTEs, where it actually executes the query on the entire dataset and _then_ joins
```sql
with my_table as (
    select *
    from read_parquet('s3://some/path/*/*/*.parquet', hive_partitioning=True)
),

some_other_table as (
  select *
  from elsewhere
)

select *
from my_table
left join some_other_table on my_table.id = some_other_table.id
where a = 'x' and b='y'
```

I checked the `GetObject` requests in the CloudWatch logs and only the subset of files required for the first couple of queries are retrieved, but every file is retrieved in the last query.

This seems like a major limitation, as queries  need to be aware of the data layout.
